### PR TITLE
Feat functions api javascript lang snippet

### PR DESCRIPTION
**Pull request by RJ025**
## Description  
added function client side snippet in web 

## Checklist  
- [x] Followed the [contribution guidelines](https://github.com/2002Bishwajeet/awesome-appwrite-snippets/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by RJ025** [comment]
@2002Bishwajeet I created a PR  could you please look into it

File: snippets/web/Funtions.code-snippets [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+
```
Comment (2002Bishwajeet) [inline snippets/web/Funtions.code-snippets:2]:
```suggestion

```

File: snippets/web/Funtions.code-snippets [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	"create Function" :{
+		"scope": "javascript , typescript",
+		"prefix": "createFunc",
+		"body": ["const ${1:response} = await ${2:function}.create(`${3:[FUNCTION_ID]}` , `${4:[name]}`)"] ,
+		"description": "Create a new function."
+	},
+
+	"create Execution": {
```
Comment (2002Bishwajeet) [inline snippets/web/Funtions.code-snippets:10]:
```suggestion
	"Create Execution": {
```

File: snippets/web/Funtions.code-snippets [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+	"Retry Build" : {
+		"scope" : "javascript , typescript" , 
+		"prefix": "retryfunc" , 
+		"body": ["const ${1:response} = await ${2:funtion}.retryBuild(`${3:[FUNCTION_ID]}` , `${4:[DEPLOYMENT_ID]}` , `${5:[BUILD_ID]}`)"] ,
+		"description": "Build unique ID."
+	}
+
+}
+
```
Comment (2002Bishwajeet) [inline snippets/web/Funtions.code-snippets:39]: Remove this white spaces please

File: snippets/web/Funtions.code-snippets [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+		"prefix": "getfunc" , 
+		"body": ["const ${1:response} = await ${2:function}.getExecution(`${3:[FUNCTION_ID]}` , `${4:[EXECUTION_ID]}`)"] ,
+		"description": "Get a function execution log by its unique ID."
+	} ,
+
+	"Retry Build" : {
+		"scope" : "javascript , typescript" , 
+		"prefix": "retryfunc" , 
+		"body": ["const ${1:response} = await ${2:funtion}.retryBuild(`${3:[FUNCTION_ID]}` , `${4:[DEPLOYMENT_ID]}` , `${5:[BUILD_ID]}`)"] ,
+		"description": "Build unique ID."
```
Comment (2002Bishwajeet) [inline snippets/web/Funtions.code-snippets:35]: description doesn't seem proper

File: snippets/web/Funtions.code-snippets [hunk 5]
```diff
@@ -0,0 +1,41 @@
+{
+
+	"create Function" :{
+		"scope": "javascript , typescript",
+		"prefix": "createFunc",
+		"body": ["const ${1:response} = await ${2:function}.create(`${3:[FUNCTION_ID]}` , `${4:[name]}`)"] ,
```
Comment (2002Bishwajeet) [inline snippets/web/Funtions.code-snippets:6]: I don't think this is an API in the client side?

**Review by 2002Bishwajeet — CHANGES_REQUESTED** [review]
Please change the filename to `functions-code.snippets` 

Also please update the description, it doesn't match exactly

**Comment by RJ025** [comment]
@2002Bishwajeet  I did the required changes could you please look into it.

File: snippets/web/Funtions.code-snippets (hunk 4 above)
Comment (RJ025) [reply snippets/web/Funtions.code-snippets]: I couldn't find any more appropriate description
could you please help me with it.

Comment (2002Bishwajeet) [reply snippets/web/Funtions.code-snippets]: @RJ025 , Remove this snippet. Seems a bug in the docs. You can rebuild a function from the client side

File: snippets/web/functions-code.snippets.code-snippets [hunk 6]
```diff
@@ -1,40 +1,28 @@
 {
-
-	"create Function" :{
-		"scope": "javascript , typescript",
-		"prefix": "createFunc",
-		"body": ["const ${1:response} = await ${2:function}.create(`${3:[FUNCTION_ID]}` , `${4:[name]}`)"] ,
```
Comment (2002Bishwajeet) [inline snippets/web/functions-code.snippets.code-snippets]: We need this snippet but the body is incorrect.

What I was expecting here is to be 
`const ${1:function} =  Function(${2:client});`

and in the description, it must be. "Create a new Function object and initialize it"

**Review by 2002Bishwajeet — CHANGES_REQUESTED** [review]
All good just one final change

File: package.json [hunk 7]
```diff
@@ -70,6 +70,10 @@
         "language": "dart",
         "path": "./snippets/flutter/teams.code-snippets"
       },
+      {
+        "language": "javascript",
+        "path": "./snippets/web/Functions.code-snippets"
```
Comment (2002Bishwajeet) [inline package.json:75]:
```suggestion
        "path": "./snippets/web/functions.code-snippets"
```

**Comment by RJ025** [comment]
@2002Bishwajeet  did the changes please check it.
and do i have to change repo readme.

File: snippets/web/functions-code.snippets.code-snippets [hunk 8]
```diff
@@ -0,0 +1,29 @@
+{
+	"Create Function" :{
+		"scope": "javascript , typescript",
+		"prefix": "createFunc",
+		"body": ["const ${1:function} = Function(${2:client})"] ,
```
Comment (2002Bishwajeet) [inline snippets/web/functions-code.snippets.code-snippets:5]:
```suggestion
		"body": ["const ${1:function} = new Functions(${2:client})"] ,
```

**Review by 2002Bishwajeet — CHANGES_REQUESTED** [review]
Yes, you have to update the Readme as well. You can look at the previous APIs of how it's done. To create a table you can use this [markdown tables generator](https://www.tablesgenerator.com/markdown_tables) to generate tables. 
 
 All we need is the `prefix` and a brief description of what it does just like the other snippets

**Comment by RJ025** [comment]
@2002Bishwajeet   I have updated readme.md and did all the required changes could you please look into it

File: README.md [hunk 9]
```diff
@@ -186,6 +186,19 @@ To know more about Health API, head over this [page](https://appwrite.io/docs/se
 | `getStorageHealth`, `getStorageLocal`                           | Check the Appwrite local storage device is up and connection is successful.                                             |
 | `getAntivirusHealth`, `getAntivirus`                            | Check the Appwrite Antivirus server is up and connection is successful.                                                 |
 
+### Functions
+
+The Functions service allows you to create custom behaviour that can be triggered by any supported Appwrite system events or by a predefined schedule.
+
+| Shortcut   | Description                                                                                    |
+|------------|------------------------------------------------------------------------------------------------|
+| `createfunc` | create a new function object and initialize it                                                 |
```
Comment (2002Bishwajeet) [inline README.md:195]:
```suggestion
| `createfunc` | Create a new function object and initialize it.                                                 |
```

File: README.md [hunk 10]
```diff
@@ -186,6 +186,19 @@ To know more about Health API, head over this [page](https://appwrite.io/docs/se
...
+| `execfunc`   | trigger a function execution, The returned object will return you the current execution status |
```
Comment (2002Bishwajeet) [inline README.md:196]:
```suggestion
| `execfunc`   | Trigger a function execution. The returned object will return you the current execution status. |
```

File: README.md [hunk 11]
```diff
@@ -186,6 +186,19 @@ To know more about Health API, head over this [page](https://appwrite.io/docs/se
...
+| `listfunc`   | Get a list of all the current user function execution logs                                     |
```
Comment (2002Bishwajeet) [inline README.md:197]:
```suggestion
| `listfunc`   | Get a list of all the current user function execution logs.                                     |
```

File: README.md [hunk 12]
```diff
@@ -186,6 +186,19 @@ To know more about Health API, head over this [page](https://appwrite.io/docs/se
...
+| `getfunc`    | Get a function execution log by its unique ID                                                  |
```
Comment (2002Bishwajeet) [inline README.md:198]:
```suggestion
| `getfunc`    | Get a function execution log by its unique ID.                                                  |
```

File: README.md [hunk 13]
```diff
@@ -186,6 +186,19 @@ To know more about Health API, head over this [page](https://appwrite.io/docs/se
 | `getStorageHealth`, `getStorageLocal`                           | Check the Appwrite local storage device is up and connection is successful.                                             |
 | `getAntivirusHealth`, `getAntivirus`                            | Check the Appwrite Antivirus server is up and connection is successful.                                                 |
 
+### Functions
+
+The Functions service allows you to create custom behaviour that can be triggered by any supported Appwrite system events or by a predefined schedule.
```
Comment (2002Bishwajeet) [inline README.md:191]:
```suggestion
The Functions service allows you to create custom behaviour that can be triggered by any supported Appwrite system events or by a predefined schedule. To know more about Functions API, head over to this [page](https://appwrite.io/docs/client/functions)
```

**Review by 2002Bishwajeet — CHANGES_REQUESTED** [review]
Final changes and then its good to go🚀

**Review by 2002Bishwajeet — APPROVED** [review]
LGTM🎇

**Comment by RJ025** [comment]
@2002Bishwajeet  i did one last change in readme

**Comment by RJ025** [comment]
is it approved @2002Bishwajeet

**Comment by 2002Bishwajeet** [comment]
Yesss. For merging it's still on hold while i double check things on my end just to be sure😉.

**Comment by 2002Bishwajeet** [comment]
Thank you for your contribution✨. Your PR has been merged 🎉

@all-contributor please add @RJ025 for code and doc

**Comment by 2002Bishwajeet** [comment]
Thank you for your contribution✨. Your PR has been merged 🎉

@all-contributors please add @RJ025 for code and doc

**Pull request merged**